### PR TITLE
#4dpzbk / - #2bj0dv - Add meta tags

### PR DESCRIPTION
**Pull request by nathakits**
- added twitter and og meta tags
- added dynamic title and descriptions meta tags
- added description text in frontmatter to md files

### CODE REVIEW

File: src/core/diy/README.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+---
+description: If you are planning on using Pupil exclusively for noncommercial purposes, and are not afraid of hacking, then we have made a guide to help you.
```
Comment (willpatera) [inline src/core/diy/README.md:2]:
```suggestion
description: If you are planning on using Pupil exclusively for noncommercial purposes, and are not afraid of hacking, then we have made a guide to help create a Pupil DIY headset.
```

File: src/core/hardware/README.md [hunk 2]
```diff
@@ -1,8 +1,12 @@
+---
+description: You wear Pupil Core headset like a pair of glasses. The headset is designed to be lightweight and adjustable in order to accommodate a wide range of users.
```
Comment (willpatera) [inline src/core/hardware/README.md:2]:
```suggestion
description: Learn about the Pupil Core eye tracking headset, adjustments, and accessories. 
```

File: src/core/software/pupil-player.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ---
 permalink: /core/software/pupil-player
+description: Pupil Player is the second tool you will use after Pupil Capture. It is a media and data visualizer at its core. You will use it to look at Pupil Capture recordings. Visualize your data and export it.
```
Comment (willpatera) [inline src/core/software/pupil-player.md:3]:
```suggestion
description: Pupil Player desktop eye tracking visualization software. Use it to play back recordings made with Pupil Core and export data for further analysis.
```

File: src/core/software/pupil-service.md [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ---
 permalink: /core/software/pupil-service
+description: Pupil Service is like Pupil Capture except it does not have a world video feed or GUI. It is intended to be used with VR and AR eye tracking setups.
```
Comment (willpatera) [inline src/core/software/pupil-service.md:3]:
```suggestion
description: Pupil Service is a GUI-less real-time eye tracking software used primarily with Pupil Labs VR/AR eye tracking add-ons. 
```

File: src/core/software/recording-format.md [hunk 5]
```diff
@@ -1,13 +1,14 @@
 ---
 permalink: /core/software/recording-format
+description: Each recording contains a number of files. The exact number and names depend on which Pupil Core headset is connected and which features are enabled during the recording.
```
Comment (willpatera) [inline src/core/software/recording-format.md:3]:
```suggestion
description: Pupil Core recording format. Learn about the files contained within a Pupil Core recording. 
```

File: src/core/terminology.md [hunk 6]
```diff
@@ -1,5 +1,6 @@
 ---
 permalink: /core/terminology
+description: In this section, we will introduce a few key terms/concepts that are essential to Pupil Core.
```
Comment (willpatera) [inline src/core/terminology.md:3]:
```suggestion
description: Key terms and concepts that are essential to Pupil Core.
```

File: src/developer/core/network-api.md [hunk 7]
```diff
@@ -1,5 +1,6 @@
 ---
 permalink: /developer/core/network-api
+description: There are two different ways that you can interact with the Pupil Core Network API. We outline how the basic communication works and how you can access the different stages of the Network API.
```
Comment (willpatera) [inline src/developer/core/network-api.md:3]:
```suggestion
description: Pupil Core has a real-time network based API. The API is language agnostic. Learn how to use the API.
```

File: src/developer/core/plugin-api.md [hunk 8]
```diff
@@ -1,5 +1,6 @@
 ---
 permalink: /developer/core/plugin-api
+description: In Pupil Core, plugins are distributed as Python files that are loaded and executed at runtime.
```
Comment (willpatera) [inline src/developer/core/plugin-api.md:3]:
```suggestion
description: Pupil Core enables end users to add their own plugins. The plugins can run and executed by Pupil Capture or Pupil Player app bundles without needing to modify source code. 
```

File: src/developer/core/recording-format.md [hunk 9]
```diff
@@ -1,5 +1,6 @@
 ---
 permalink: /developer/core/recording-format
+description: We recommend reading the format thourougly if you want to develop software that creates recordings on its own, or processes existing recordings without having to open them in Pupil Player.
```
Comment (willpatera) [inline src/developer/core/recording-format.md:3]:
```suggestion
description: Pupil Core recording format. Learn more about the files included in a Pupil Core recording. 
```

File: src/invisible/hardware/README.md [hunk 10]
```diff
@@ -1,3 +1,7 @@
+---
+description: Add eye tracking powers to your HTC Vive with our 200Hz binocular eye tracking add-on.
```
Comment (willpatera) [inline src/invisible/hardware/README.md:2]:
```suggestion
description: Learn about your Pupil Invisible glasses features and accessories. 
```

File: src/invisible/user-guide/analysis.md [hunk 11]
```diff
@@ -1,5 +1,6 @@
 ---
 permalink: /invisible/user-guide/analysis
+description: There are different options available for analyzing recordings made with Pupil Invisible. You can use Pupil Cloud, Pupil Player, or export recordings to other third-party tools.
```
Comment (willpatera) [inline src/invisible/user-guide/analysis.md:3]:
```suggestion
description: Replay, visualize, and analyze eye tracking recordings made with Pupil Invisible. You can use Pupil Cloud, Pupil Player, or work with some of our third party partners. 
```

File: src/invisible/user-guide/invisible-monitor.md [hunk 12]
```diff
@@ -1,6 +1,10 @@
+---
+description: Pupil Invisible Monitor is a stand alone desktop app. It shows all available Pupil Invisible Devices in your Local Area Network.
```
Comment (willpatera) [inline src/invisible/user-guide/invisible-monitor.md:2]:
```suggestion
description: Pupil Invisible Monitor is a stand alone desktop app used to stream scene video and gaze data from all available Pupil Invisible eye tracking devices on your local WiFi network. 
```

File: src/invisible/user-guide/troubleshooting.md [hunk 13]
```diff
@@ -1,5 +1,6 @@
 ---
 permalink: /invisible/user-guide/troubleshooting
+description: In this section are for troubleshooting.
```
Comment (willpatera) [inline src/invisible/user-guide/troubleshooting.md:3]:
```suggestion
description: Need assistance? Check out this section for troubleshooting with Pupil Invisible. 
```

File: src/core/README.md [hunk 14]
```diff
@@ -1,3 +1,7 @@
+---
+description: Welcome to Pupil Core! The first thing you need to do is download the latest Pupil Core software.
```
Comment (willpatera) [inline src/core/README.md:2]:
```suggestion
description: Welcome to Pupil Core! This section will guide you through setup (software and hardware) in order for you to make your first recording with Pupil Core. 
```

File: src/core/academic-citation.md [hunk 15]
```diff
@@ -1,5 +1,7 @@
 ---
 permalink: /core/academic-citation
+description: Please take a look at our papers below for citation options. If you're using Pupil as a tool in your research please cite the below UbiComp 2014 paper.
```
Comment (willpatera) [inline src/core/academic-citation.md:3]:
```suggestion
description: Writing an academic paper that uses Pupil Labs products? Please cite us!
```

**Review by willpatera — CHANGES_REQUESTED** [review]
@nathakits some changes for descriptions.

**Comment by nathakits** [comment]
@willpatera All changes are accepted

**Pull request merged**